File: Universal Windows Platform Solution/Phantom of the West/Phantom of the West/Visual Novel/Chapters/Chapter 9/IWillPetFoxtrot.cs
using Phantom_of_the_West.Visual_Novel.Serialization.Event_Flagging;

namespace Phantom_of_the_West.Visual_Novel.Chapters.Chapter_9
{
	internal class IWillPetFoxtrot : StoryChoice
	{
		bool kickedCheerleader = (PotWVN.MainVN.EventFlags.Ch5SparredWithCheerleader == CheerleaderSpar.Kicked);

		internal IWillPetFoxtrot() : base(StoryChoice.GetLocalizedString("ch9IWillPetFoxtrot"), 9096)
		{
			if (kickedCheerleader)
			{
				NextState = 9095;
			}
		}

		public override void Select()
		{
			FoxtrotReaction reaction;
			if (kickedCheerleader)
			{
				reaction = FoxtrotReaction.Hissed;
			}
			else
			{
				reaction = FoxtrotReaction.AcceptedPet;
			}
			PotWVN vn = PotWVN.MainVN;
			vn.EventFlags.Ch9FoxtrotReacted = reaction;
			base.Select();
		}
	}
}

File: Android Project/app/src/main/java/com/christopherraleigh/phantomofthewest/visual_novel/chapters/chapter_9/IWillPetFoxtrot.java
package com.christopherraleigh.phantomofthewest.visual_novel.chapters.chapter_9;

import com.christopherraleigh.phantomofthewest.R;
import com.christopherraleigh.phantomofthewest.visual_novel.PotWVN;
import com.christopherraleigh.phantomofthewest.visual_novel.chapters.StoryChoice;
import com.christopherraleigh.phantomofthewest.visual_novel.serialization.event_flagging.FoxtrotReaction;

/**
 * Created by christopherraleigh on 2016-05-21.
 */
public class IWillPetFoxtrot extends StoryChoice {
    private FoxtrotReaction reaction;

    private IWillPetFoxtrot(int nextState, FoxtrotReaction reaction) {
        super(R.string.ch9IWillPetFoxtrot, nextState);
        this.reaction = reaction;
    }

    public static IWillPetFoxtrot Create(boolean kickedCheerleader) {
        int nextState;
        FoxtrotReaction reaction;
        if (kickedCheerleader) {
            nextState = 9095;
            reaction = FoxtrotReaction.Hissed;
        } else {
            nextState = 9096;
            reaction = FoxtrotReaction.AcceptedPet;
        }
        return new IWillPetFoxtrot(nextState, reaction);
    }

    @Override
    public void select() {
        PotWVN vn = PotWVN.getMainVN();
        vn.getEventFlags().setCh9FoxtrotReacted(reaction);
        super.select();
    }
}
